### PR TITLE
feat: notify site admins on campaign send failure

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Notify site admin if a campaign fails to send.

See 1200550061930446-as-1205525012341463/f

### How to test the changes in this Pull Request:

1. Set Mailchimp as the ESP
1. Throw a wrench in the works by changing the MC API URL [here](https://github.com/Automattic/newspack-newsletters/blob/dbc9ac1d9b57882b6eae0a91aa4bc3e5a633ba84/includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php#L820)
2. Try to send a campaign
3. Observe an email about the failure is sent to the site administrator's email

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/service-providers/class-newspack-newsletters-service-provider.php [hunk 1]
```diff
@@ -336,6 +336,24 @@ public function insert_post_data( $data, $postarr ) {
 		return $data;
 	}
 
+	/**
+	 * Notify the site administrators.
+	 *
+	 * @param string $subject The email subject.
+	 * @param string $message The email message.
+	 */
+	private static function notify_site_admins( $subject, $message ) {
```
Comment (chickenn00dle) [inline includes/service-providers/class-newspack-newsletters-service-provider.php:345]: I think sending these emails as well as selecting how many and which accounts are notified should be tied to some option or hook to allow publishers to opt-in or modify this. Forcing every publisher to start receiving failure emails seems a bit extreme.

File: includes/service-providers/class-newspack-newsletters-service-provider.php [hunk 2]
```diff
@@ -336,6 +336,24 @@ public function insert_post_data( $data, $postarr ) {
...
+		$site_administrators = get_users(
+			[
+				'role'  => 'administrator',
+				'limit' => 10, // Notifying 10 people should be enough.
+			]
+		);
+		foreach ( $site_administrators as $user ) {
+			$sent = \wp_mail( $user->user_email, $subject, $message ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_mail_wp_mail
```
Comment (chickenn00dle) [inline includes/service-providers/class-newspack-newsletters-service-provider.php:353]: This is the email I receive when my newsletter fails:

Screenshot 2024-02-27 at 17 03 37

It feels a bit bare and doesn't really provide me with too much information. We might want to include some additional details like:

- Some details about which newsletter failed
- related to the first point, maybe a link to their newsletter or newsletter settings in wp-admin
- Maybe even a support link if they can't figure out how to address this on their end

**Review by chickenn00dle — COMMENTED** [review]
Hey @adekbadek,

This is working as expected, but I had some concerns around allowing users to opt-in or customize as well as with the email content. Left some comments.

File: includes/service-providers/class-newspack-newsletters-service-provider.php (hunk 1 above)
Comment (adekbadek) [reply includes/service-providers/class-newspack-newsletters-service-provider.php]: Updated in cfa11e6255df7be8427ea4af6e44b11a3bde6ab0 to use the email listed as the `admin_email` option.

File: includes/service-providers/class-newspack-newsletters-service-provider.php (hunk 2 above)
Comment (adekbadek) [reply includes/service-providers/class-newspack-newsletters-service-provider.php]: Updated in cfa11e6255df7be8427ea4af6e44b11a3bde6ab0 to include campaign title and edit link.

**Review by chickenn00dle — APPROVED** [review]

**Pull request merged**